test/integration/targets/etcd3/tasks/main.yml

```diff
@@ -0,0 +1,144 @@
+---
+# test code for the etcd3 module
+# (c) 2017,  Jean-Philippe Evrard <jean-philippe@evrard.me>
+
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+# ============================================================
+
+- name: Check if etcd3ctl exists for re-use.
+  shell: "ETCDCTL_API=3 {{ etcd3_path }}/etcdctl --endpoints=localhost:2379 get foo"
+  args:
+    executable: /bin/bash
+  changed_when: false
+  failed_when: false
+  register: _testetcd3ctl
+
+# Installing etcd3
+- name: If can't reuse, prepare download folder
+  file:
+    path: "{{ etcd3_download_location }}"
+    state: directory
+  register: _etcddownloadexists
+  when:
+    - _testetcd3ctl.rc != 0
+
+- name: Delete download folder if already exists (to start clean)
+  file:
+    path: "{{ etcd3_download_location }}"
+    state: absent
+  when:
+    - _testetcd3ctl.rc != 0
+    - _etcddownloadexists is not changed
+
+- name: Recreate download folder if purged
+  file:
+    path: "{{ etcd3_download_location }}"
+    state: directory
+  when:
+    - _testetcd3ctl.rc != 0
+    - _etcddownloadexists is not changed
+
+- name: Download etcd3
+  unarchive:
+    src: "{{ etcd3_download_url }}"
+    dest: "{{ etcd3_download_location }}"
+    remote_src: yes
+  when:
+    - _testetcd3ctl.rc != 0
+
+# Install requirements for the etcd3 module
+  pip:
+    name: etcd3
+    state: present
+
+# Running etcd3 and kill afterwards if it wasn't running before.
+- name: Run etcd3
+  shell: "{{ etcd3_path }}/etcd &"
+  register: _etcd3run
+  changed_when: true
+  when:
+    - _testetcd3ctl.rc != 0
+
+# Integration tests
+- name: Check mode, show need change
+  etcd3:
+    key: "foo"
+    value: "bar"
+    state: "present"
+  register: _etcd3_prst_chktst
+  check_mode: true
+- name: Change to new value
```

Comment:
Could you please add a blank line between tasks, will make it a little more readable

Reply:
Good catch.